libs/s3/s3_versioning_common_test_lib.py

```diff
@@ -377,25 +380,37 @@ def upload_versions(s3_test_obj: S3TestLib, s3_ver_test_obj: S3VersioningTestLib
     return versions
 
 
-# pylint: disable=too-many-arguments
-def upload_version(s3_test_obj: S3TestLib, bucket_name: str, object_name: str,
-                   file_path: str, versions_dict: dict,
-                   chk_null_version: bool = False, is_unversioned: bool = False) -> None:
-    """ Upload objects to a versioning enabled/suspended bucket and return dictionary of uploaded
-    versions
+def upload_version(s3_test_obj: Union[S3TestLib, S3MultipartTestLib], bucket_name: str,
+                   object_name: str, file_path: str, versions_dict: dict, **kwargs) -> None:
+    """ Upload an object(Multipart/Regular) to a versioning enabled/suspended bucket and return
+    dictionary of uploaded versions.
 
     :param s3_test_obj: S3TestLib object to perform S3 calls
     :param bucket_name: Bucket name for calling PUT Object
     :param object_name: Object name for calling PUT Object
     :param file_path: File path that can be used for PUT Object call
-    :param chk_null_version: True, if 'null' version id is expected, else False
     :param versions_dict: Dictionary to be updated with uploaded version metadata
-    :param is_unversioned: Set to true if object is uploaded to an unversioned bucket
+    :keyword chk_null_version: True, if 'null' version id is expected, else False
+    :keyword is_unversioned: Set to true if object is uploaded to an unversioned bucket
         Can be used for setting up pre-existing objects before enabling/suspending bucket
         versioning
+    :keyword is_multipart: True if upload version of object is multipart
+    :keyword total_parts: Total number of parts used in multipart upload
+    :keyword file_size: Size of the object, multiple of 1MB
     """
-    res = s3_test_obj.put_object(bucket_name=bucket_name, object_name=object_name,
-                                 file_path=file_path)
+    chk_null_version = kwargs.get("chk_null_version", False)
+    is_unversioned = kwargs.get("is_unversioned", False)
+    is_multipart = kwargs.get("is_multipart", False)
+    total_parts = kwargs.get("total_parts", 2)
+    file_size = kwargs.get("file_size", 10)
+    if is_multipart:
+        res = s3_test_obj.complete_multipart_upload_with_di(
```

Comment:
@apoorva-rao This will increase code complexity and number of paramters required for complete multipart api so I would recomend to have it in seperate method. we can discuss this if required.